swift-sdk/Internal/InAppManager.swift

```diff
@@ -104,6 +149,26 @@ class InAppManager : NSObject, IterableInAppManagerProtocolInternal {
             self.lastSyncTime = self.dateProvider.currentDate
         }
     }
+    
+    func set(read: Bool, forMessage message: IterableInAppMessage) {
```

Comment:
Should we call `onInboxChanged` here, when read state changes?

Reply:
We are using onInboxChanged for passing changes coming from the server to the client. I don't think this will be helpful because the client that will set the value to read will get a onInboxChanged callback again. 
I can see the value if different parts of the client iOS app is using this callback. But for that a notification is more appropriate.

Reply:
Now we have notification. I have added the call to fire this notification.